lib/rules/color-hex-case/index.js

```diff
@@ -14,6 +15,9 @@ const messages = ruleMessages(ruleName, {
 	expected: (actual, expected) => `Expected "${actual}" to be "${expected}"`,
 });
 
+const HEX = /^#[0-9A-Za-z]+/;
```

Comment:
Shall we use RegExp, which looks for valid HEX only?

```suggestion
const HEX = /^#([\da-f]{3,4}|[\da-f]{6}|[\da-f]{8})$/i;
```

https://regexr.com/5k70k

![CleanShot 2021-01-13 at 22 49 53](https://user-images.githubusercontent.com/654597/104514602-d1c8a900-55f1-11eb-8695-3033a064c629.png)

Reply:
My goal was to match the previous style search (and not break existing tests) for this one, and the original query was:

```
/^#[0-9A-Za-z]+/
```

But if we're ok with a potentially "breaking" change that will only look at valid hex colors, I'm ok to make that.

Reply:
I would say it's not a breaking change, but a bugfix. What do you think, @jeddy3?

Reply:
I believe the rule was originally designed to check the case of anything that _sort of looks like a hex colour_, i.e. a word that starts with hash and is followed by numbers and letters, as:

1. people may want to come up with their own hex-like languages extensions
2. we already had the [color-no-invalid-hex rule](https://stylelint.io/user-guide/rules/color-no-invalid-hex) rule the people could use to check for hex colour validity against the specs

As it's more extensible, and because users may already be relying on this behaviour, I suggest we keep it as.